server/src/test/java/org/apache/druid/server/http/CoordinatorCompactionConfigsResourceTest.java

```diff
@@ -407,6 +412,115 @@ public void testAddOrUpdateCompactionConfigWithoutExistingConfig()
+                     newConfigCaptor.capture(),
+                     ArgumentMatchers.any()
+                 )
+    ).thenReturn(ConfigManager.SetResult.ok());
+
+    final DataSourceCompactionConfig newConfig = new DataSourceCompactionConfig(
+        "dataSource",
+        null,
+        500L,
+        null,
+        new Period(3600),
+        null,
+        new UserCompactionTaskGranularityConfig(Granularities.HOUR, null, null),
+        null,
+        null,
+        null,
+        null,
+        null,
+        ImmutableMap.of("key", "val")
+    );
+    Response ignore = coordinatorCompactionConfigsResource.addOrUpdateCompactionConfig(
+        newConfig,
+        mockHttpServletRequest
+    );
+    Assert.assertEquals(null, newConfigCaptor.getValue().getCompactionConfigs().get(0).getEngine());
+  }
+
+  @Test
+  public void testAddOrUpdateCompactionConfigWithInvalidMaxNumTasksForMSQEngine()
+  {
+    Mockito.when(mockConnector.lookup(
+                     ArgumentMatchers.anyString(),
+                     ArgumentMatchers.eq("name"),
+                     ArgumentMatchers.eq("payload"),
+                     ArgumentMatchers.eq(CoordinatorCompactionConfig.CONFIG_KEY)
+                 )
+    ).thenReturn(null);
+    Mockito.when(mockJacksonConfigManager.convertByteToConfig(
+                     ArgumentMatchers.eq(null),
+                     ArgumentMatchers.eq(CoordinatorCompactionConfig.class),
+                     ArgumentMatchers.eq(CoordinatorCompactionConfig.empty())
+                 )
+    ).thenReturn(CoordinatorCompactionConfig.empty());
+    final ArgumentCaptor<byte[]> oldConfigCaptor = ArgumentCaptor.forClass(byte[].class);
+    final ArgumentCaptor<CoordinatorCompactionConfig> newConfigCaptor = ArgumentCaptor.forClass(
+        CoordinatorCompactionConfig.class);
+    Mockito.when(mockJacksonConfigManager.set(
+                     ArgumentMatchers.eq(CoordinatorCompactionConfig.CONFIG_KEY),
+                     oldConfigCaptor.capture(),
+                     newConfigCaptor.capture(),
+                     ArgumentMatchers.any()
+                 )
+    ).thenReturn(ConfigManager.SetResult.ok());
+
+    int maxNumTasks = 1;
+
+    final DataSourceCompactionConfig newConfig = new DataSourceCompactionConfig(
+        "dataSource",
+        null,
+        500L,
+        null,
+        new Period(3600),
+        null,
+        new UserCompactionTaskGranularityConfig(Granularities.HOUR, null, null),
+        null,
+        null,
+        null,
+        null,
+        CompactionEngine.MSQ,
+        ImmutableMap.of(ClientMSQContext.CTX_MAX_NUM_TASKS, maxNumTasks)
+    );
+    Response response = coordinatorCompactionConfigsResource.addOrUpdateCompactionConfig(
+        newConfig,
+        mockHttpServletRequest
+    );
+    Assert.assertEquals(DruidException.Category.INVALID_INPUT.getExpectedStatus(), response.getStatus());
+    Assert.assertEquals(
+        StringUtils.format(
```

Comment:
No need to format messages in tests, especially when the format args (`maxNumTasks`) are coming from the test itself. Compare against the full formatted message itself.